mautic/mautic
# Role export access in Leads, Forms, Reports

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Added new permission for export in form results. This PR not require migrations, exports are default enable. This permission give ability disable export permission for Contacts, Forms, Reports.

![image](https://user-images.githubusercontent.com/462477/39305083-21ff9dc0-495c-11e8-9e23-1805dd639c8e.png)

#### Steps to test this PR:
1.  Create new user with role and enable all permission for Contacts, Forms, Reports with unchecked new Export access: Disable 
2.  Login as new user and see If exports buttons exists and exports works properly in all sections: Contacts (batch export), Form (results exports) and Reports
3. Open HTML exports of Form results and Reports.
4. Edit in incognito user permission and disable export 
5. See If export button are hidden in Form results and  Reports and Contacts lists in batch.
6. See If opened HTML export from 3. return access denied after refresh


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/5995"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
